doc/how-to-analyze-haskell-code-performance.rst

```diff
@@ -0,0 +1,164 @@
+which runs in the browser and can open this JSON file directly.
+See the
+`Haskell Optimization Handbook <https://haskell.foundation/hs-opt-handbook.github.io>`__
+on how to optimize your code based on the profiling results afterwards.
+
+Different report formats can be generated by using different RTS flags. Some useful ones are:
+
+- ``-p`` for a GHC's own
+  `standard report <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#cost-centres-and-cost-centre-stacks>`__
+  ``<app-name>.prof``, which can be visualized with `profiteur <https://github.com/jaspervdj/profiteur>`__
+  or `ghcprofview <https://github.com/portnov/ghcprofview-hs>`__.
+- ``-pj`` for a
+  `JSON report <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#json-profile-format>`__
+  ``<app-name>.prof``, which can be visualized with `Speedscope <https://speedscope.app>`__.
+- ``-l-p`` for a binary
+  `"eventlog" report <https://downloads.haskell.org/ghc/latest/docs/users_guide/runtime_control.html#rts-eventlog>`__
+  ``<app-name>.eventlog``, which contains a lot more details and can show you resource usage over time, and can
+  be converted to JSON with `hs-speedscope <https://github.com/mpickering/hs-speedscope>`__
+  and visualized with `Speedscope <https://speedscope.app>`__.
+
+.. note::
+
+    The report generated from the ``cabal configure``, ``cabal build`` etc. commands can
+    also be generated with a single ``cabal run`` command, but no settings are persisted
+    to a file:
+
+    .. code-block:: console
+
+          $ cabal run --enable-profiling --profiling-detail=late my-app -- +RTS -pj -RTS
+          <program builds, runs and finishes>
+
+
+You can also pass a lot more
+`profiling compiler options <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#compiler-options-for-profiling>`__
+to GHC via the ``ghc-prof-options`` field in your ``my-app.cabal`` file.
+Add the following, recommended profiling options to your ``my-app.cabal`` file
+to control where performance measuring code is inserted into your application:
+
+::
+
+    executable my-app
+      ...
+      ghc-prof-options:
+        -fprof-auto
+        -fno-prof-count-entries
+        -fprof-auto-calls
+        ...
+
+You can find more information and further options in the
+`GHC "cost-center" guide <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#automatically-placing-cost-centres>`__.
+
+Profiling your dependencies too
+-------------------------------
+
+The setup so far only profiles your main application, which is usually what you want.
+This happens by default, because Cabal command line options only apply to local packages
+and dependencies are usually not local.
+However, the bottlenecks may be in your dependencies, so you would want to profile those too.
+
+First, to enable ``late``-cost-center profiling of all packages/dependencies in your project,
+add the following to your project’s ``cabal.project`` file:
```

Comment:
Should it keep using `cabal.project.local`? I think one file is a better idea than two. Especially given the argument you provide when introducing `.local`: `.project` are meant for Gitting, while `.local` are not.

Reply:
Let's keep it simple with a single file.
Btw, i couldn't find any mentioning of `cabal.prof.project` xD.

Reply:
> Let's keep it simple with a single file.

So, `cabal.projct.local`, right? I still see `cabal.project`...

Reply:
Fixed